File: frontend/src/types/Order.ts
import type { User } from './User';
import type { OrderItem } from './OrderItem';
import type { Payment } from './Payment';
import type { Return } from './Return';
import type { Shipment } from './Shipment';
import type { OrderStatusHistory } from './OrderStatusHistory';
import type { RecipientInformation } from './RecipientInformation';

export interface Order {
  id: number;
  userId: number;
  status: string;
  subtotal: number;
  totalAmount: number;
  notes: string;
  discountAmount: number;
  shippingFee: number;
  estimateDeliveryFrom: string;
  estimateDeliveryTo: string;
  createdAt: string;
  updatedAt: string;
  user?: User;
  orderItems?: OrderItem[];
  payment?: Payment;
  returnInfo?: Return;
  shipment?: Shipment;
  orderStatusHistory?: OrderStatusHistory;
  recipientInformation?: RecipientInformation;
}

File: backend/src/main/java/iuh/fit/backend/model/Order.java
package iuh.fit.backend.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

// Order Entity
@Entity
@Table(name = "orders")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "user_id")
    private Integer userId;

    private String status;

    private BigDecimal subtotal;

    @Column(name = "total_amount")
    private BigDecimal totalAmount;

    private String notes;

    @Column(name = "coupon_id")
    private Integer couponId;

    @Column(name = "coupon_code", length = 50)
    private String couponCode;

    @Column(name = "discount_amount")
    private BigDecimal discountAmount;

    @Column(name = "shipping_fee")
    private BigDecimal shippingFee;

    @Column(name = "estimate_delivery_from")
    private LocalDate estimateDeliveryFrom;

    @Column(name = "estimate_delivery_to")
    private LocalDate estimateDeliveryTo;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @ManyToOne
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    @JsonIgnore
    private User user;

    @OneToMany(mappedBy = "order", fetch = FetchType.LAZY)
    @JsonIgnore
    private List<OrderItem> orderItems;

    @OneToOne(mappedBy = "order", fetch = FetchType.LAZY)
    @JsonIgnore
    private Payment payment;

    @OneToOne(mappedBy = "order", fetch = FetchType.LAZY)
    @JsonIgnore
    private Return returnInfo;

    @OneToOne(mappedBy = "order", fetch = FetchType.LAZY)
    @JsonIgnore
    private Shipment shipment;

    @OneToMany(mappedBy = "order", fetch = FetchType.LAZY)
    @JsonIgnore
    private List<OrderStatusHistory> orderStatusHistories;

    @OneToOne(mappedBy = "order", fetch = FetchType.LAZY)
    @JsonIgnore
    private RecipientInformation recipientInformation;

    @ManyToOne
    @JoinColumn(name = "coupon_id", insertable = false, updatable = false)
    @JsonIgnore
    private Coupon coupon;
}
